File: UI/pc-part-picker/src/model/Harddrive.ts
import { HarddriveType } from "./HarddriveType";
import { ManufacturerType } from "./ManufacturerType";
import { PcPart } from "./Pcpart";

export interface Harddrive extends PcPart {
    capacity: number | null;
    pricePerGb: number | null;
    type: string | null;
    cache: number | null;
    interface: string | null;
    manufacturerTypeId: number | null;
    driveTypeId: number | null;
    driveType: string | null;
    manufacturerType: string | null;
}

File: API/PCPartPicker/Models/Harddrive.cs
using System;
using System.Collections.Generic;

namespace PCPartPicker.Models;

public partial class Harddrive
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public decimal? Price { get; set; }

    public decimal? Capacity { get; set; }

    public decimal? PricePerGb { get; set; }

    public string? Type { get; set; }

    public decimal? Cache { get; set; }

    public string? Interface { get; set; }

    public int? ManufacturerTypeId { get; set; }

    public int? DriveTypeId { get; set; }

    public virtual ICollection<Configuration> Configurations { get; set; } = new List<Configuration>();

    public virtual HarddriveType? DriveType { get; set; }

    public virtual ManufacturerType? ManufacturerType { get; set; }
}
